src/rdkafka_fetcher.c

```diff
@@ -599,7 +737,11 @@ rd_kafka_fetch_reply_handle_partition(rd_kafka_broker_t *rkb,
                 rd_kafka_aborted_txns_destroy(aborted_txns);
         if (likely(rktp != NULL))
                 rd_kafka_toppar_destroy(rktp); /*from get()*/
-        rd_kafka_buf_skip_tags(rkbuf);
+
+        if (rd_kafka_buf_ApiVersion(request) >= 12 &&
+            hdr.ErrorCode == RD_KAFKA_RESP_ERR_NO_ERROR)
+                rd_kafka_buf_skip_tags(rkbuf);
+
```

Comment:
Here it reads the partition tags
```suggestion
        if (PartitionTags) {
                /* Set default LeaderId and LeaderEpoch */
                PartitionTags->CurrentLeader.LeaderId    = -1;
                PartitionTags->CurrentLeader.LeaderEpoch = -1;
        }
        rd_kafka_buf_read_tags(rkbuf,
                               rd_kafkap_Fetch_reply_tags_partition_parse,
                               TopicTags, PartitionTags);

```

Reply:
Fixed